### PR TITLE
feat(TG-74): implement statistics storage

### PR DESCRIPTION
**Pull request by kolesnik**
basic implementation with data logging per each event.  Work is still in progress here. Let's not merge it to master

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-backend/pull/68" target="_blank">ReviewNinja

File: src/devices/device-statistic.model.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
```
Comment (velmyk) [inline src/devices/device-statistic.model.js]: ES 6 modules use strict mode by default. Please remove this declaration.

Comment (kolesnik) [inline src/devices/device-statistic.model.js]: Agree

File: src/devices/device-statistic.model.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Device Statistic Schema
+ */
+
+// NOTE: we can consider to use capped collection to limit the size of statistic collection
+// see - http://mongoosejs.com/docs/guide.html#capped and https://docs.mongodb.com/manual/core/capped-collections/
+
+// NOTE: we are using object _id as timestamp property
+// for additional details see - https://steveridout.github.io/mongo-object-time/ , http://stackoverflow.com/questions/8749971/can-i-query-mongodb-objectid-by-date
+const DeviceStatisticSchema = new mongoose.Schema({
+    'event': {
```
Comment (borys-ihnatyev) [inline src/devices/device-statistic.model.js]: We should add this lint rule ```quote-props: ["error", "as-needed"]```
http://eslint.org/docs/rules/quote-props

File: src/devices/device-statistic.model.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        'type': String
+    },
+    'value': {
+        'type': Object
+    }
+});
+
+// TODO: Consider index options
+
+export default mongoose.model('Device_Statistic', DeviceStatisticSchema);
```
Comment (borys-ihnatyev) [inline src/devices/device-statistic.model.js]: Dash case ?))

File: src/devices/newDeviceHandler.test.js [hunk 4]
```diff
@@ -33,7 +33,8 @@ describe('# New device handler', () => {
         sut = proxyquire('./newDeviceHandler', {
             '../data-streams/input': input,
             '../data-streams/output': output,
-            '../API/sensors/sensor.model': Sensor
+            '../API/sensors/sensor.model': Sensor,
+            './event/event-type': { DEVICE_STATUS: 'status'}
```
Comment (borys-ihnatyev) [inline src/devices/newDeviceHandler.test.js]: Replace please ```'status'``` with declared event type above

```js
import {DEVICE_STATUS} from './event/event-type';
// ...
'./event/event-type': { DEVICE_STATUS }
// ...
}
//...

**Comment by kolesnik** [comment]
Closed due an other implementation

**Pull request closed without merging**